### PR TITLE
fix: 修复 Carousel 组件 left/right 模式下卡片轮播方向

### LINKED ISSUE #20780: bug: carousel 组件卡片轮播方向
- [x] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[https://next.ant.design/components/carousel-cn/](https://next.ant.design/components/carousel-cn/)
ref: #20645
### Steps to reproduce
- 选择 left 或 right 
- 点击切换卡片
- 卡片轮播方向为左右

### What is expected?
- 卡片轮播方向为上下

### What is actually happening?
- 卡片轮播方向为左右

| Environment | Info |
|---|---|
| antd | 4.0.0-rc.0 |
| React | The latest |
| System | mac |
| Browser | chrome |

### PR DESCRIPTION
**Pull request by xrkffgg**
### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link
close https://github.com/ant-design/ant-design/issues/20780

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |  Fixed Carousel card carousel orientation in left / right mode         |
| 🇨🇳 Chinese |修复 Carousel 组件 left/right 模式下卡片轮播方向           |

### ☑️ Self Check before Merge

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by yoyo837** [comment]
Please add some test case for this patch.

**Comment by xrkffgg** [comment]
@yoyo837 不知道为啥 4.0-rc 把关于 `vertical` 这部分都给去了，但确实是影响了原有的效果

**Comment by yoyo837** [comment]
#20547 @afc163

File: components/carousel/index.tsx [hunk 1]
```diff
@@ -105,11 +114,12 @@ export default class Carousel extends React.Component<CarouselProps, {}> {
     const prefixCls = getPrefixCls('carousel', props.prefixCls);
     const dotsClass = 'slick-dots';
     const dotPosition = this.getDotPosition();
+    props.vertical = dotPosition === 'left' || dotPosition === 'right';
```
Comment (afc163) [inline components/carousel/index.tsx]: 保留这一行就行了。

Comment (xrkffgg) [reply components/carousel/index.tsx]: OK

**Pull request merged**